src/rdpc/analysesProcessor.ts

```diff
@@ -215,6 +310,117 @@ export const getAllMergedDonor = async ({
   return mergedDonors;
 };
 
+export const getAllMergedDonorWithSpecimens = async ({
+  analysesFetcher = fetchAnalysesWithSpecimens,
+  egoJwtManager,
+  studyId,
+  url,
+  config,
+  donorIds,
+}: {
+  studyId: string;
+  url: string;
+  egoJwtManager: EgoJwtManager;
+  donorIds?: string[];
+  config?: {
+    chunkSize?: number;
+    state?: StreamState;
+  };
+  analysesFetcher: typeof fetchAnalysesWithSpecimens;
+}): Promise<SpecimensByDonors> => {
+  let mergedDonors: SpecimensByDonors = {};
```

Comment:
You use `let` for the documents you want to keep updating the contents of but you can use `const` in these cases. You never re-assign the variable to a new object, you only modify the contents of that object, so `const` is better.